### DISCUSSION TITLE
Is the MQTT Device Service for Jakarta production-ready?

### DISCUSSION THREAD

**Discussion by dmocek** [edgexfoundry/community#146] [General]
Is the MQTT Device Service for Jakarta production-ready?

**Comment by JamesKButcher** [comment]
The MQTT Device Service at [EdgeX 2.1 / Jakarta](https://wiki.edgexfoundry.org/pages/viewpage.action?pageId=70948690) forms part of our community ‘Long Term Support’ (LTS) list of services. As described in the [EdgeX LTS policy](https://wiki.edgexfoundry.org/pages/viewpage.action?pageId=69173332), this means that the community will provide support and fixes/updates for issues that are deemed critical or highly significant. But do note this is not a commercial agreement and support comes on a best effort basis from our technical community of volunteers.

So based on the LTS agreement, we believe the MQTT Device Service works well and at least to the level of functionality as described by its documentation at EdgeX 2.1. But companies perhaps need to make their own decisions about what “Production Ready” means, e.g., each may have their own definitions for metrics like performance, throughput, reliability and so on.

Note that commercial companies are sometimes able to offer enhanced versions of the EdgeX services. If I may mention my own company, IOTech, here… we provide commercial support for the full EdgeX stack but have also implemented/enhanced more Device Services that can run against open-source EdgeX. In the case of MQTT we have added additional features and mechanisms as described in our [IOTech product docs](https://docs.iotechsys.com/edge-xpert23/device-services/mqtt/overview/mqtt-overview.html).

**Comment by JamesKButcher** [comment]
Oh yes, let me also mention that the LTS support for 2.1 / Jakarta will cease in November 2023. At that point we should have [EdgeX 3.1 / Napa](https://wiki.edgexfoundry.org/display/FA/Napa+Release) released, which will take on the role of the LTS version. So good to look to begin or at least consider upgrading to EdgeX V3.x fairly soon, if indeed the LTS aspect is important for you

**Comment by dmocek** [comment]
Thanks @JamesKButcher
